13_biblioteki/13_8_AlgorithmsBLAS.tex

```diff
@@ -0,0 +1,257 @@
+					T = A(K,J) - SDOT(K-1, A(1,K), 1,
+					A(1,J), 1)
+					T = T/A(K,K)
+					A(K,J) = T
+					S = S + T*T
+			10 CONTINUE
+			20 CONTINUE
+			S = A(J,J) - S
+			... EXIT
+			IF (S .LE. 0.0E0) GO TO 40 A(J,J) = SQRT(S)
+			30 CONTINUE		
+		\end{lstlisting}
+	\end{frame}
+	\begin{frame}{Realizacja z użyciem BLAS level 1 I}
+		\begin{tabular}{ l l l }
+			\textbf{FUNCTION} & \textunderscore \textbf{DOT(N,X, INCX,Y,INCY) = dot} & $ \leftarrow x^T * y $ \\
+			   & $\uparrow$ & $ \leftarrow x^H * y $ \\
+			   & S &  \\
+			   & D & \\
+			   & DS & \\
+		\end{tabular} \\ 
+		\textbf{N} - number of elements of vectores x and y to be used in dot \\
+		\textbf{X} - array of length $lenx = (n - 1) * |inc x| + 1$ containing the n-vector $x$ \\
+		\textbf{INCX} - increment for the array $x$: \\
+		$\textit{incx} \geq 0 $ $x$ stored forward in array $x$, i.e: $x_i \rightarrow x((i - 1) * incx + 1)$ \\
+		$\textit{incx} < 0$ $x$ stored backward in array $x$, i.e: $x_i \rightarrow x((i - n) * incx + 1)$ \\
+		\textbf{Y} - \\
+		\textbf{INCY} - \\
+	\end{frame}
+	\begin{frame}{Realizacja z użyciem BLAS level 1 II}
+		OUTPUT: $s = \sum_{i=1}^n x_i * y_i$, $s = \sum_{i=1}^n x_i^{*} * y_i$ \\
+		if $incx = 0 \Rightarrow x_i = x(1) \forall i$ \\
+		$incy = 0 \Rightarrow y_i = y(1) \forall i$ \\
+		S = SDOT(n, x, incx, y, incy)
+	\end{frame}
+	\begin{frame}[allowframebreaks]{Realizacja z użyciem BLAS level 2}
+	\begin{tabular}{ l l l }
+		\textunderscore \textbf{TRSV} & \textbf{(UPLO, TRANS, DIAG, N, A, LDA, X, INCX)} \\
+		   $\uparrow$ & \\
+		   S, D, C, Z &  \\
+	\end{tabular} \\ 
+	we: $ Ay = x $, wy: $x \leftarrow y$ \\
+	\textit{Solve Triangular System}: $ x \leftarrow A^{-1}x $\\ 
+	$ A^{-1} = (A^T)^{-1}, A^{-H} = (A^H)^{-1} $ 
+	\begin{tabular}{c c}
+	$
+	\left .
+ 		 \begin{tabular}{c}
+			 $ x \leftarrow A^{-T} x $ \\
+			 $ x \leftarrow A^{-H} x $ 
+  		\end{tabular}
+	\right \}
+	$
+	 &
+		\begin{tabular}{c}
+			$ A \leftarrow A^T $ \\
+			$ A \leftarrow A^H $
+  		\end{tabular} 
+	\end{tabular} \\
+	
+	\textbf{A} - nxn upper or lower - triangular matrix A \\
+	\textbf{uplo} - upper/lower triangular opt \\
+	= 'L' - solve lower triangular system (forward elimin) \\
+	= 'U' - solve upper triangular system (back substitutuion) \\
+	trans - transposition option for A: \\
+	$N - : x \leftarrow A^{-1} x;$ \\
+	$T - : x \leftarrow A^{-T} x;$ \\
+	$C - : x \leftarrow A^{-H} x$ \\
+	\textbf{diag} - whether A is unit triangular i.e. $a_{ii} = 1$ \\
+	N - diag of A stored in array; \\
+	U - diag = 1, \\
+	\textbf{n} - no of rows and columns in A \\
+	\textbf{A} - array containing n-by-n triangular matrix \\
+	\textbf{lda} - the leading dimension of array a (decl. in call p) $lda \geq max(n, 1)$
+	\textbf{x} - array of $lenx = (n - 1) * |incx| + 1 \leftarrow$ rhs $n$-vector $x$ \\
+	\textbf{incx} - increment for x
+	OUTPUT:  $x$ - the solution vector replaces the input \\
+	Notes: subprogram do not check singularity of A \\
+	\end{frame}
```

Comment:
I think some indentation would be great to make the above frames more readable. Maybe make itemized lists or put each of A, uplo, trans, etc.. in definition blocks? Because now it's a wall of text :(